packages/client-sdk-web/src/internal/vector-index-data-client.ts

```diff
@@ -70,19 +80,60 @@ export class VectorIndexDataClient implements IVectorIndexDataClient {
         item.setVector(vector);
 
         item.setMetadataList(
-          vectorIndexItem.metadata === undefined
-            ? []
-            : Object.entries(vectorIndexItem.metadata).map(([key, value]) => {
-                const metadata = new vectorindex._Metadata();
-                metadata.setField(key);
-                metadata.setStringValue(value);
-                return metadata;
-              })
+          VectorIndexDataClient.convertItemMetadataToProtobufMetadata(
+            vectorIndexItem
+          )
         );
         return item;
       })
     );
+    return request;
+  }
 
+  private static convertItemMetadataToProtobufMetadata(
+    item: VectorIndexItem
+  ): vectorindex._Metadata[] {
+    if (item.metadata === undefined) {
+      return [];
+    }
+    return Object.entries(item.metadata).map(([key, value]) => {
+      const metadata = new vectorindex._Metadata();
+      metadata.setField(key);
+      if (typeof value === 'string') {
+        metadata.setStringValue(value);
+      } else if (typeof value === 'number') {
+        if (Number.isInteger(value)) {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+          metadata.setIntegerValue(value);
```

Comment:
I am unclear why some of these modifiers have unsafe call warnings and others do not.

Reply:
I pulled this down locally to test this. I don't think that you need any of these `eslint-disable` lines actually. The js protobuf dependecies trip up some IDEs. I suspect that your IDE is operating off of some stale data, and that it thinks that `item` is of type `any` or `unknown`. I think if you remove all of the `eslint-disable` lines, and rebuild in your terminal it will pass. (Verified this :onmybox: ). Sometimes if you rebuild on your terminal, _and then_ open up the project inside your IDE, it will pickup the latest changes. :skeletor_fist_shake: _caching issues_ :skeletor_fist_shake: